admin/plugins.php

```diff
@@ -0,0 +1,411 @@
+<?php
+/**
+ * Admin settings helper functions.
+ *
+ * @package performance-lab
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Gets plugin info from plugin slug.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $plugin_slug The string identifier for the plugin in questions slug.
+ *
+ * @return array Array of plugin data, or empty if none/error.
+ */
+function perflab_get_plugin_info( string $plugin_slug ) {
+	$plugin = plugins_api(
+		'plugin_information',
+		array(
+			'slug'   => $plugin_slug,
+			'fields' => array(
+				'short_description' => true,
+				'icons'             => true,
+			),
+		)
+	);
+
+	if ( is_wp_error( $plugin ) ) {
+		return array();
+	}
+
+	if ( is_object( $plugin ) ) {
+		$plugin = (array) $plugin;
+	}
+
+	return $plugin;
+}
+
+/**
+ * Gets info about all the standalone plugins and their status.
+ *
+ * @since n.e.x.t
+ *
+ * @return array of plugins info.
+ */
+function perflab_get_standalone_plugins() {
+	$managed_standalone_plugins = array(
+		'webp-uploads'            => array(
+			'plugin_data' => array(),
+		),
+		'performant-translations' => array(
+			'plugin_data' => array(),
+		),
+		'dominant-color-images'   => array(
+			'plugin_data' => array(),
+		),
+	);
+
+	foreach ( $managed_standalone_plugins as $managed_standalone_plugin_slug => $managed_standalone_plugin_entry ) {
+		$managed_standalone_plugins[ $managed_standalone_plugin_slug ]['plugin_data'] = perflab_get_plugin_info( $managed_standalone_plugin_slug );
+	}
+
+	return $managed_standalone_plugins;
+}
+
+/**
+ * Renders plugin UI for managing standalone plugins within PL Settings screen.
+ *
+ * @since n.e.x.t
+ */
+function perflab_render_plugins_ui() {
+	require_once ABSPATH . 'wp-admin/includes/plugin-install.php';
+	require_once ABSPATH . 'wp-admin/includes/plugin.php';
+
+	$standalone_plugins = perflab_get_standalone_plugins();
+
+	if ( empty( $standalone_plugins ) ) {
+		return;
+	}
+	?>
+	<div class="wrap">
+		<h1><?php esc_html_e( 'Performance Plugins', 'performance-lab' ); ?></h1>
+		<p><?php esc_html_e( 'The following standalone performance plugins are available for installation.', 'performance-lab' ); ?></p>
+		<div class="wrap">
+			<form id="plugin-filter" method="post">
+				<div class="wp-list-table widefat plugin-install wpp-standalone-plugins">
+					<h2 class="screen-reader-text"><?php esc_html_e( 'Plugins list', 'default' ); ?></h2>
+					<div id="the-list">
+						<?php
+						foreach ( $standalone_plugins as $standalone_plugin ) {
+							perflab_render_plugin_card( $standalone_plugin );
+						}
+						?>
+					</div>
+				</div>
+			</form>
+		</div>
+		<div class="clear"></div>
+	</div>
+	<?php
+}
+
+/**
+ * Renders individual plugin cards.
+ *
+ * @since n.e.x.t
+ *
+ * @param array $standalone_plugin Plugin data as passed from get_standalone_plugins().
+ */
+function perflab_render_plugin_card( array $standalone_plugin ) {
+	$plugin = $standalone_plugin['plugin_data'];
```

Comment:
It's a bit odd to expect an array and then extract out the one piece of information from that array. Why not simply expect that `plugin_data` array in the first place? It makes the function a bit more flexible and decouples from the exact shape that that data needs to be "packaged" within the overall array.

Maybe rename the parameter/variable to `$plugin_data` to clarify?

Reply:
@felixarntz good call, I've now passed only the `plugin_data` array, and also renamed usage of `$plugin` to be `$plugin_data` in that function block, as it seems more appropriate.

This change has been effected in https://github.com/WordPress/performance/pull/864/commits/f43d47ccdd5f820c918337d0213dcc853aacdceb